packages/dnb-eufemia/src/extensions/forms/Iterate/ArrayPushButton.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useCallback, useContext } from 'react'
+import classnames from 'classnames'
+import { Button } from '../../../components'
+import { ButtonProps } from '../../../components/Button'
+import { forwardSpaceProps } from '../utils'
+import IterateElementContext from './IterateElementContext'
+import { useDataValue } from '../hooks'
+import {
+  ComponentProps,
+  DataValueReadProps,
+  DataValueWriteProps,
+} from '../types'
+
+export type Props = ComponentProps &
+  DataValueReadProps<unknown[]> &
+  DataValueWriteProps<unknown[]> & {
+    text?: ButtonProps['text']
+    pushValue: unknown
+    children?: ButtonProps['children']
+  }
+
+function ArrayPushButton(props: Props) {
+  const iterateElementContext = useContext(IterateElementContext)
+  const { handlePush } = iterateElementContext ?? {}
+
+  const { text, value, pushValue, handleChange, children } =
```

Comment:
```suggestion
  const { text, value, pushValue, handleChange, children, ...rest } =
```

So we can send along HTML attributes like aria-*.

Reply:
This could mix things up a bit, since `useDataValue` adds a few optional props and callbacks, such as `onFocus`, which would then end up being sent to `Button`. There is also optional callbacks like `handleChange` and `onChange` in separate callbacks (explained in code comments) which would both be sent to `Button`. I would suggest extending some function to pick area-tags or similar sets of properties like it is done for spacing props here, to ensure only intended props are passed into sub components.

Reply:
Kind of a discussion between whitelisting and blacklisting

Reply:
From my experience, we NEED to allow everything to be forwarded to the button. In that case we need to know all the properties from a nested component and remove them, or destruct them.

Reply:
We could forward everything, but then it would have to be from the received props (direct to the component, not those received through `useDataValue`), and then override the specific props from `useDataValue` that we know it will have overridden, like the click event. Otherwise the Button will receive a lot of props from logic inside `useDataValue` that is not made for the Button component. It might ignore them, but it might be similar names intended fir different use cases, which could lead to problems. However, if the ArrayPushButton extend the props of Button, and pass them directly (not through the hook) and just use specific props from the hook, it will not cause that problem.

Reply:
Yes! Sure, I that's right. I probably did comment on the wrong place about the rest props 🙏

Reply:
Fixed now ✅